### DISCUSSION TITLE
[Noob] Installation on new Pi Zero 2 W stuck

### DISCUSSION THREAD

**Discussion by poughkeepsee** [mKeRix/room-assistant#953] [General] [answered]
Hi!

Pretty new to the world of Room Assistant. I bought a new RPi Zero 2 W and I'm trying to follow the Pi Zero instructions, but I keep getting stuck during the installation when running ```sudo npm i --global --unsafe-perm room-assistant```

I tried two times and waited more than 1h on each try only to see the process had gotten stuck on different stages. Pi was still connected to wifi and all else seemed good.

This got me thinking, since the RPi 2 W has a different processor than the old Zero (actually as I understand it the 2 W is more similar to the old RPi 3) , will there need to be new instructions for the 2 W? Or can I use the instructions for the RPi 3/4?

Or is the case that I may be just messing something up and the old Zero instructions should be working?

Thanks all :)

**Comment by poughkeepsee** [comment]
Update: I've tried using the installation instructions for the Pi 3/4 and it worked flawless. Guess this answers it.

**Reply by dennis-bell** [reply]
I can confirm the same issues with the RPI Zero 2W


```sudo npm i --global --unsafe-perm room-assistant --loglevel verbose```

```npm timing build:run:install:node_modules/room-assistant/node_modules/@mkerix/noble Completed in 42858ms
npm info run epoll@4.0.1 install node_modules/room-assistant/node_modules/epoll node-gyp rebuild
npm info run @abandonware/bleno@0.5.1-4 install { code: 0, signal: null }
npm timing build:run:install:node_modules/room-assistant/node_modules/bleno Completed in 42918ms
npm info run i2c-bus@5.2.2 install node_modules/room-assistant/node_modules/i2c-bus node-gyp rebuild
client_loop: send disconnect: Broken pipe run i2c-bus@5.2.2 install node_modules/room-assistant/node_modules/i2c-bus node-gyp rebuild```

**Reply by Segar85** [reply]
> Update: I've tried using the installation instructions for the Pi 3/4 and it worked flawless. Guess this answers it.

Hi imcass,

I am also using Pi Zero 2 W as well and face the same issues. I tried the Pi 3/4 instruction as you mention but my gets stucked under : reify:mathjs: http fetch GET 200 https://registry.npmjs.org/mathjs/-/mathjs-9.5.2.tgz 104370ms (cache miss)

Will you be able to advise how to rectify the issue?

**Reply by poughkeepsee** [reply]
> > Update: I've tried using the installation instructions for the Pi 3/4 and it worked flawless. Guess this answers it.
> 
> Hi imcass,
> 
> I am also using Pi Zero 2 W as well and face the same issues. I tried the Pi 3/4 instruction as you mention but my gets stucked under : reify:mathjs: http fetch GET 200 https://registry.npmjs.org/mathjs/-/mathjs-9.5.2.tgz 104370ms (cache miss)
> 
> Will you be able to advise how to rectify the issue?

Hi, @Segar85 

I’m by no means an expert here, I tried using the regular Pi 3/4 method and worked for me, but I don’t have the expertise to troubleshoot your issue. 

Did you try using the regular Pi Zero instructions as well? Assuming the Dev may have possibly updated it.

**Reply by Segar85** [reply]
I managed to solve my issue.

**Comment by 1liminal1** [comment, marked as answer]
Are there any answers to this? My install is also getting stuck.

Thanks

**Reply by Segar85** [reply]
Are you using Raspberry Pi Zero or Zero 2 W.

Since I am using the Zero 2 W, i followed the guide that was meant for Rasp Pi 3. 

However when i used this command to do the install : `sudo npm i --global --unsafe-perm room-assistant` i kept getting error.

After much digging and googling around i found that using this command `sudo npm i --global --unsafe-perm --verbose --foreground-scripts room-assistant` I was able to install without errors..

So give this command a try and continue following the rest of the installation guide commands.

**Reply by 1liminal1** [reply]
Its a Pi Zero W (OG)

And thank you! That worked. maybe an issue with SSH...  Not sure, but thanks for that :)

**Reply by davem2421** [reply]
> 

@Segar85 you're a lifesaver!! thanks so much for posting this. I was driving myself nuts trying to figure out what I was doing wrong!
